.github/actions/deploy/deploy.py

```
@@ -0,0 +1,1334 @@
+            ['kubectl', 'create', 'namespace', self.deployment_namespace],
+            check=False  # Don't fail if namespace already exists
+        )
+
+        # Initialize TLS manager for kind clusters after namespace setup
+        if self.args.pod_to_pod_tls_enabled:
+            self.tls_manager = TLSCertificateManager(self)
+            print('🔐 TLS certificate manager initialized for podToPodTLS')
+
+    def run_command(self,
+                    cmd: List[str],
+                    cwd: str = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)